src/main/java/seedu/address/logic/commands/task/AddTaskCommand.java

```diff
@@ -37,38 +35,41 @@
     public static final String MESSAGE_ADD_TASK_SUCCESS = "New Task added to Room %1$d. \nDescription: %2$s";
 
     private final Task taskToAdd;
-    private final Index roomIndex;
+    private final int roomNumber;
 
     /**
      * Creates an AddTaskCommand to add the specified {@code Task} to a {@code Room}
-     * with the {@code roomIndex}.
+     * with the {@code roomNumber}.
      */
-    public AddTaskCommand(Task task, Index roomIndex) {
-        requireAllNonNull(task, roomIndex);
+    public AddTaskCommand(Task task, int roomNumber) {
+        requireNonNull(task);
+        assert roomNumber > 0 : "Room number should be greater than 0.";
```

Comment:
i think you should be asserting if the `roomNumber` is `null`? Because if the roomNumber is <0 then it should be the user fault and not the system fault no?

Reply:
`roomNumber` cannot actually be `null` though, because it is a primitive (i.e. `int`).